parlai/core/metrics.py

```diff
@@ -523,6 +529,60 @@ def compute_many(
         )
 
 
+class IntraDistinctMetric(AverageMetric):
+    """
+    Compute intra-distinct (per-utterance).
+    """
+
+    @classmethod
+    def _ngram(cls, seq, n: int):
+        for i in range(len(seq) - n + 1):
```

Comment:
as discussed, I'm a bit less certain about the scenario when `len(seq) < n`. Others LGTM

Reply:
Manually checked and this sequence doesn't give partial n-grams. I think this implementation is faithful to the original.